<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->

  <meta name="description" content="Holden Lee's Research Notebook">
  <meta name="author" content="Holden Lee">
    
  <title>The Blessing and the Curse of the Multiplicative Updates</title>

  <!-- Bootstrap core CSS -->
  <link href="../../../../bootstrap/css/bootstrap.min.css" rel="stylesheet">

  <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
  <!-- <link href="../../assets/css/ie10-viewport-bug-workaround.css" rel="stylesheet"> -->

  <!-- Custom styles for this template -->
  <link href="../../../../css/blog.css" rel="stylesheet">
  <link href="../../../../css/default.css" rel="stylesheet">

  <!-- Extension : Footnotes -->
  <link href="../../../../footnotes/css/footnotes.css" rel="stylesheet">

  <!-- Extension : InlineDisqussions @ https://github.com/tsi/inlineDisqussions -->
  

  <!-- Extension : Collapsible lists @ http://code.stephenmorley.org/javascript/collapsible-lists/-->
  <link href="../../../../collapsible_lists/css/collapsible.css" rel="stylesheet">
  <script type="text/javascript" src="../../../../collapsible_lists/js/CollapsibleLists.js"></script>

</head>

<body>

<!-- Navigation bar. navbar-inverse is black, navbar-default is white.-->
<!-- To make a button active (pressed), use <li class="active"> -->
<div id="header">
  <nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../../../../">Notebook</a>
      </div>
      <div id="navbar" class="collapse navbar-collapse">
        <ul class="nav navbar-nav">
          <li><a href="../../../../">Home</a></li>
          <li><a href="../../../../sitemap.html">Sitemap</a></li>
<!-- TODO: Distinguish between PAPERS, RESEARCH QUESTIONS, BOOKS -->
<!-- TODO: make this part a for loop over main pages -->
        </ul>
      </div><!--/.nav-collapse -->
    </div>
  </nav>
</div>
<!-- Content -->
<!--div id="content">
  <h1>Mental Wilderness</h1>-->



<div class="container">
  <div id="content">
    <div class="page header">
      <h1>The Blessing and the Curse of the Multiplicative Updates</h1>
    </div>
    <div class="info">
      
       
        <p>Posted: 2017-02-12 
          , Modified: 2017-02-12 
	</p>
      
       <p>Tags: <a href="../../../../tags/multiplicative%20update.html">multiplicative update</a>, <a href="../../../../tags/online%20learning.html">online learning</a>, <a href="../../../../tags/biology.html">biology</a>, <a href="../../../../tags/evolution.html">evolution</a></p> 
    </div>
    
  </div>
  <!--/div-->

  <div class="toc"> <ul>
 <li><a href="#dispelling-the-curse">Dispelling the curse</a></li>
 <li><a href="#summary">Summary</a></li>
 </ul> </div>

  <div class="blog-main">
    <p>(Chapter by Manfred Warmuth)</p>
<p>Setting: online learning</p>
<p>Maintain weights where <span class="math inline">\(w_i\)</span> is the belief that the <span class="math inline">\(i\)</span>th expert is best. Prediction: weighted average.</p>
<p>Multiplicative updates motivated by minimum relative entropy principle. Simplest is Bayes’s rule for updating priors.</p>
<blockquote>
<p>The “blessing” of these updates is that they learn very fast in the short term because the good weights grow exponentially. However, their “curse” is that they learn too fast and wipe out other weights too quickly. This loss of variety can have a negative effect in the long term because adaptivity is required when there is a change in the process generating the data stream.</p>
</blockquote>
<blockquote>
<p>Surprisingly, some of Nature’s methods parallel the ones developed in machine learning, but Nature also has some additional tricks.</p>
</blockquote>
<p>Evolution on 2 scales:</p>
<ul>
<li>Short-term: mutation and selection (multiplicative update)</li>
<li>Long-term: stability - requires mechanism for preventing quick convergence to currently fittest strategy or gene.</li>
</ul>
<p><span class="math display">\[
\wt s_i = \fc{s_ie^{-\eta \text{loss}_i}}{Z}
= \amin_{r\in \R^n, \sum_i r_i=1} \sum_i KL(r||s) + \eta\sum_i r_il_i
\]</span> Relative entropy keeps <span class="math inline">\(\wt s\)</span> close to <span class="math inline">\(s\)</span>. <span class="math inline">\(\eta\)</span> is tradeoff parameter. <span class="math inline">\(W_i = e^{-\eta l_i}\)</span> is the fitness of strategy <span class="math inline">\(i\)</span>. Bayes Rule: expected log loss, <span class="math inline">\(\eta=1\)</span>.</p>
<p>(systematic way of deriving such updates by trading off a relative entropy between the new and old weight vector with a loss function that measures how well the weights did on the current instance vector)</p>
<p>Why multiplicative, not additive update? Converge quickly when data stream generating process is static.</p>
<p>Problem: data changes over time. MU loss of variety.</p>
<p>Use of MU: separate good/bad RNA, reamplify good. Loop. Use PCR.</p>
<p>Assumption: <span class="math inline">\(W_i\)</span> independent of share vector <span class="math inline">\(s\)</span>. (Otherwise: apostatic effects.) <span class="math inline">\(\wt s_i = \fc{w_iW_i}{\an{s,W}}\)</span>.</p>
<p>In-vitro selection: iterate Bayes with same data likelihoods.</p>
<blockquote>
<p>“brittle” because the gradients of the losses appear in the exponents of the update factors. This is problematic when there is noise in the data and the data-generating process changes over time</p>
</blockquote>
<p>Compounded: weights have constant precision.</p>
<p>Curse is extinctino in nature.</p>
<h3 id="dispelling-the-curse">Dispelling the curse</h3>
<p>Ex. Bacteria in nutriet solution separate into 3 niches, 3 species. Mixing kills all but one.</p>
<blockquote>
<p>three species of bacteria that play an RPS game. When started on a Petri dish, colonies of each species develop that slowly chase each other around the dish: R invades S’s colonies, S invades P, and P invades R</p>
</blockquote>
<ol type="1">
<li>Niche boundaries help prevent the curse. (Human: multiplicative update of meme shares causes loss of variety.)
<ul>
<li>Key problem 1 (that cannot be solved by MU alone): There are three strands of RNA in a tube and the goal is to amplify the mixture while keeping the percentages of the strands unchanged. This is to be done without sequencing the strands and determining their concentrations.
<ul>
<li>Solution: Translate to (double stranded) DNA and using an enzyme, add a specific short “end strand” to both ends of all strands in the tube. These end strands function as connectors between strands and make it possible to randomly ligate many strands together into long strands. Now separate out one long strand. With the help of an enzyme, add “primer strands” to both ends of that long strand. Apply PCR iteratively starting with the long selected strand, always making complete copies of the same original long strand that is located between the primers. Stop when you have the target amount of DNA. Now divide the long strands into their constituents by cutting them at the specific end strand that functioned as the connector. Finally, remove all short primer and end strands and convert back to RNA.</li>
<li>The long strand functions as a “chromosome.” Free floating genes in the nuclei of cells would compete.</li>
</ul></li>
</ul></li>
<li>Coupling preserves diversity.
<ul>
<li>Genes on chromosome selected for together; genes “cooperate” for sake of efficient copying.</li>
<li>What updates can be implemented with in-vitro selection/blind computation? Can you have negative weights?</li>
<li><span class="math inline">\(EG^{\pm}\)</span> algorithm: maintain 2 weight vectors <span class="math inline">\(s^+,s^-\in \R^n\)</span>. Label <span class="math inline">\((s^+-s^-)\cdot x\)</span>, loss <span class="math inline">\(L((s^+-s^-)\cdot x,y) = ((s^+-s^-_)\cdot x-y)^2\)</span> so that fitness is inverse.</li>
<li>Problem 2: find small set of RNA strands that can bind to <span class="math inline">\(q\)</span> different protein sheets. (Assume that among all strands in a 1-liter tube of RNA, there is a particular set of two strands such that for each of q proteins, at least one of the two has a high fraction of attachment. Can you use PCR to get <span class="math inline">\(\approx (0.5, 0.5, 0,\ldots)\)</span>?)
<ul>
<li>First idea: cycle through <span class="math inline">\(q\)</span> proteins and do in-vitro selection.</li>
<li>Problem: overselecting can kill off one.</li>
<li>ML problem: <span class="math inline">\(\exists u\)</span> with <span class="math inline">\(k\)</span> nonzeros of value <span class="math inline">\(\rc k\)</span>, <span class="math inline">\(u\cdot W_j\ge \rc k\)</span>. Find <span class="math inline">\(s\)</span> with <span class="math inline">\(s\cdot W_j \ge \rc{2k}\)</span>.</li>
<li>Normalized winnow algorithm: If <span class="math inline">\(s\cdot W_j\ge \rc{2k}\)</span>, then <span class="math inline">\(\wt s=s\)</span> (conservative update), else apply MU.</li>
</ul></li>
<li>ML 1: Prevent over-training by making the update conservative.</li>
<li>in the context of in-vitro selection this means that RNA strands that attached to the protein sheet are simply recombined into the tube without the PCR amplification step.</li>
<li>the amplification step occurs at most <span class="math inline">\(O(k\log \fc Nk)\)</span> times if there is a consistent k-literal disjunction and this is information theoretically optimal. Grows logarithmically in <span class="math inline">\(n\)</span>.</li>
<li>If select for <span class="math inline">\(\binom nk\)</span> combinations, not necessary to be conservative. But this is too large.</li>
<li>Coupling can be replaced by thresholding.</li>
<li>Cap shares/weight from above.</li>
</ul></li>
<li>Super (apex) predator.
<ul>
<li>Nibbles at the highest bar of the histogram of possible prey species. (Removing it causes this species to take over.) The more frequent a species, the more opportunity for a disease to spread and this can keep a species from taking over.</li>
<li>ML2: Cap the weights from above for the purpose of learning a set of experts.</li>
<li>After MU: all weights that exceed c are reduced to c and the total weight gained is redistributed among the remaining components that lie below c so that their ratios are preserved and the total weight still sums to one. (cf. exploration, <span class="math inline">\(\ep\)</span>-greedy.)</li>
<li>capping solves a constrained optimization problem.</li>
<li>Ex. constrain to be in convex hull of <span class="math inline">\(k\)</span>-corners. This still only needs <span class="math inline">\(n\)</span> weights.</li>
<li>In matrix MU, share vectors are density matrices - prob vector of eigenvalues</li>
<li>quantum relative entropy instead of the regular relative entropy</li>
<li>MMU with capping on eigenvalues: PCA. <span class="math inline">\(k\)</span> experts become <span class="math inline">\(k\)</span>-dim subspace. Capped density matrices are convex combs of <span class="math inline">\(k\)</span>-dim subspaces.</li>
<li>relating: lower bound shares.</li>
<li>keep a batch of the initial mixture in reserve. When too uniform, add some!</li>
<li>Disk spindown problem.</li>
<li><span class="math inline">\(\wt s = (1-\al) s^m + \al \rc N\one\)</span>.</li>
</ul></li>
<li>Mutations keep a base variety.
<ul>
<li>Data shifts once in a while and some are a return back to data seen previously. Need:</li>
<li>Capability to bring up shares quickly.</li>
<li>Remember experts that did well in past. (Sleeping)</li>
<li>ML4: Sleeping realizes long-term memory.</li>
<li>Keep track of average share vector, mix in <span class="math inline">\(\al\)</span> of <span class="math inline">\(r\)</span>.</li>
<li>Markov network with 2 tracks (tubes).</li>
<li>It is reasonable to expect that Nature makes use of the sleeping mechanism as well. (Seed sprouting.)</li>
</ul></li>
</ol>
<p>How is sleeping realized at genetic level. Junk DNA, sex?</p>
<h2 id="summary">Summary</h2>
<p>Machine Learning mechanisms:</p>
<ol type="1">
<li>conservative update for learning multiple goals</li>
<li>upper bounding weights for learning multiple goals</li>
<li>lower bounding weights for robustness against change</li>
<li>sleeping for realizing a long-term memory.</li>
</ol>
<p>Nature’s mechanisms:</p>
<ol type="1">
<li>coupling for preserving variety</li>
<li>boundaries for preserving variety</li>
<li>super-predators for preserving variety</li>
<li>mutations for keeping a base variety.</li>
</ol>

  </div>

    

    <!-- Extension : Sharing buttons @ www.sharethis.com -->
    <span class="st_facebook_large" displayText="Facebook"></span>
    <span class="st_twitter_large" displayText="Tweet"></span>
    <span class="st_googleplus_large" displayText="Google +"></span>
    <span class="st_reddit_large" displayText="Reddit"></span>
    <span class="st__large" displayText></span>

    <div id="disqus_thread"></div>
    


  
    <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>
  </div>
  
</div>


<!-- Footer -->
<div id="footer">
  <div class="container">
    Built with
    <a href="http://jaspervdj.be/hakyll">Hakyll</a> 
    using 
    <a href="http://www.getbootstrap.com">Bootstrap</a>, 
    <a href="http://www.disqus.com">Disqus</a>,
    <a href="http://ignorethecode.net/blog/2010/04/20/footnotes/">Footnotes.js</a>,
    <a href="http://highlightjs.org/">Highlight.js</a>, 
    <a href="http://www.mathjax.org">MathJax</a>, 
    and <a href="http://www.sharethis.com">ShareThis</a>.
  </div>
</div>
</body>

</html>

<!-- SCRIPTS -->
<!-- jQuery-->
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
<script src="http://code.jquery.com/jquery-1.10.1.min.js"></script>

<script src="../../../../bootstrap/js/bootstrap.min.js"></script>

<!-- Extension : Highlight.js @ https://highlightjs.org/ -->
<!-- Syntax highlighting tomorrow-night-bright, agate-->
<link rel="stylesheet" href="../../../../highlight/css/tomorrow-night-bright.css">
<script src="../../../../highlight/js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<!-- Extension : MathJax @ https://docs.mathjax.org/en/v2.5-latest/tex.html -->
<!-- MathJax/config/local/local.js contains macros. Need to provide entire URL-->
<script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML,http://holdenlee.github.io/notebook/MathJax/config/local/local"></script>

<!-- Extension : Footnotes @ http://ignorethecode.net/blog/2010/04/20/footnotes/ -->
<script src="../../../../footnotes/js/footnotes.js"></script>

<!-- Extension : Disqus @ http://disqus.com -->
<!-- Extension : InlineDisqussions @ https://github.com/tsi/inlineDisqussions -->

<script src="../../../../disqus/js/disqus.js"></script>



<!-- Extension : Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-73261814-1', 'auto');
  ga('send', 'pageview');

</script>

